### ISSUE TITLE
openstack v3 authentication for unscoped token fails with: TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by toabctl** [apache/libcloud#1365]
## Summary

I try to authenticate against an internal OpenStack cloud that is based on OpenStack Rocky. 

## Detailed Information

My reproducer script is:
```
#!/usr/bin/python
from libcloud.compute.types import Provider
from libcloud.compute.providers import get_driver
import libcloud.security

libcloud.security.VERIFY_SSL_CERT = False

OpenStack = get_driver(Provider.OPENSTACK)
o = OpenStack('myusername', 'PASSWORD',
              ex_force_auth_url='https://keystone.local:5000/v3/auth/tokens',
              ex_tenant_name='cloud',
              ex_token_scope='unscoped',
              ex_force_service_region='CustomRegion',
              ex_domain_name='ldap_users',
              ex_force_auth_version='3.x_password').list_images()
```

Running that with the latest libcloud version from pypi (currently 2.6.0) in a virtual env results in:
```
$ PYTHONPATH=. python ~/tmp/test-libcloud.py 
/home/tom/devel/apache/libcloud/venv-2.7/lib/python2.7/site-packages/urllib3-1.25.6-py2.7.egg/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
Traceback (most recent call last):
  File "/home/tom/tmp/test-libcloud.py", line 17, in <module>
    ex_force_auth_version='3.x_password').list_images()
  File "/home/tom/devel/apache/libcloud/libcloud/compute/drivers/openstack.py", line 372, in list_images
    self.connection.request('/images/detail').object, ex_only_active)
  File "/home/tom/devel/apache/libcloud/libcloud/common/openstack.py", line 225, in request
    raw=raw)
  File "/home/tom/devel/apache/libcloud/libcloud/common/base.py", line 538, in request
    action = self.morph_action_hook(action)
  File "/home/tom/devel/apache/libcloud/libcloud/common/openstack.py", line 292, in morph_action_hook
    self._populate_hosts_and_request_paths()
  File "/home/tom/devel/apache/libcloud/libcloud/common/openstack.py", line 334, in _populate_hosts_and_request_paths
    auth_version=self._auth_version)
  File "/home/tom/devel/apache/libcloud/libcloud/common/openstack_identity.py", line 185, in __init__
    service_catalog=service_catalog)
  File "/home/tom/devel/apache/libcloud/libcloud/common/openstack_identity.py", line 428, in _parse_service_catalog_auth_v3
    for item in service_catalog:
TypeError: 'NoneType' object is not iterable
```

The reason is that an unscoped v3 auth request does not return a service catalog. See https://docs.openstack.org/api-ref/identity/v3/?expanded=password-authentication-with-unscoped-authorization-detail,password-authentication-with-explicit-unscoped-authorization-detail,get-service-catalog-detail,password-authentication-with-scoped-authorization-detail#authentication-and-token-management

**Comment by kshtsk** [comment]
Probable fix is #1367

**Comment by toabctl** [comment]
This was fixed via https://github.com/apache/libcloud/pull/1367
